adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/OAuthRequestAuthenticator.java

```diff
@@ -375,6 +381,23 @@ protected String stripOauthParametersFromRedirect() {
                 .replaceQueryParam(OAuth2Constants.STATE, null);
         return builder.build().toString();
     }
-
+    
+    private String rewrittenRedirectUri(String originalUri) {
+        String rewrittenRedirectUri = originalUri;
```

Comment:
Creating a URL instance is relatively costly (believe it or not) so I'd suggest first checking if deployment.getRedirectRewriteRules() is null and just return originalUri straight away if it is